x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/config_panel_wrapper.tsx

```diff
@@ -52,19 +67,223 @@ export const ConfigPanelWrapper = memo(function ConfigPanelWrapper(props: Config
+              aria-label={i18n.translate('xpack.lens.xyChart.addLayerButton', {
+                defaultMessage: 'Add layer',
+              })}
+              title={i18n.translate('xpack.lens.xyChart.addLayerButton', {
+                defaultMessage: 'Add layer',
+              })}
+              onClick={() => {
+                dispatch({
+                  type: 'UPDATE_STATE',
+                  subType: 'ADD_LAYER',
+                  updater: state =>
+                    appendLayer({
+                      activeVisualization,
+                      generateId,
+                      trackUiEvent,
+                      activeDatasource: datasourceMap[activeDatasourceId],
+                      state,
+                    }),
+                });
+              }}
+              iconType="plusInCircleFilled"
+            />
+          </EuiToolTip>
+        </EuiFlexItem>
+      )}
+    </EuiForm>
+  );
+}
+
+function LayerPanel(
+  props: ConfigPanelWrapperProps &
+    VisualizationLayerConfigProps<unknown> & {
+      isOnlyLayer: boolean;
+      activeVisualization: Visualization;
+      onRemove: () => void;
+    }
+) {
+  const { framePublicAPI, layerId, activeVisualization, isOnlyLayer, onRemove } = props;
+  const datasourcePublicAPI = framePublicAPI.datasourceLayers[layerId];
+  const layerConfigProps = {
+    layerId,
+    dragDropContext: props.dragDropContext,
+    state: props.visualizationState,
+    setState: props.setState,
+    frame: props.framePublicAPI,
+  };
+
+  return (
+    <EuiPanel className="lnsConfigPanel__panel" paddingSize="s">
+      <EuiFlexGroup gutterSize="s" alignItems="flexStart" responsive={false}>
+        <EuiFlexItem grow={false}>
+          <LayerSettings
+            layerId={layerId}
+            isOnlyLayer={isOnlyLayer}
+            layerConfigProps={layerConfigProps}
+            activeVisualization={activeVisualization}
+            onRemove={onRemove}
+          />
+        </EuiFlexItem>
+
+        {datasourcePublicAPI && (
+          <EuiFlexItem className="eui-textTruncate">
+            <NativeRenderer
+              render={datasourcePublicAPI.renderLayerPanel}
+              nativeProps={{ layerId }}
+            />
+          </EuiFlexItem>
+        )}
+      </EuiFlexGroup>
+
+      <EuiSpacer size="s" />
+
+      <NativeRenderer
+        render={activeVisualization.renderLayerConfigPanel}
+        nativeProps={layerConfigProps}
+      />
+
+      <EuiSpacer size="s" />
+
+      <EuiFlexItem>
+        <EuiButtonEmpty
+          size="xs"
+          iconType="trash"
+          color="danger"
+          data-test-subj="lns_layer_remove"
+          onClick={onRemove}
+        >
+          {isOnlyLayer
+            ? i18n.translate('xpack.lens.resetLayer', {
+                defaultMessage: 'Reset layer',
+              })
+            : i18n.translate('xpack.lens.deleteLayer', {
+                defaultMessage: 'Delete layer',
+              })}
+        </EuiButtonEmpty>
```

Comment:
There's something weird about the focus state on this button. It stays focused after getting clicked. My guess is that it's related to the click handler.

Reply:
Generally, buttons stay focused when you click them, unless you explicitly write code to prevent that behavior. I agree, though, the UX in this case is strange. I had to write some surprisingly hacky code to unfocus the button, though.

Reply:
Why would keeping the focus on a button that was just clicked be weird UX? Where would you expect the focus to move to?

Reply:
The button is a "clear" button, why would we want to keep focus after clearing once?

Reply:
Instead of blurring the focus of the element which forces the user's focus back to the very first element with in the DOM (Global navigation), this button should be disabled if the layer is completely clear/empty. This way it no longer "looks" focused but when it goes from enabled to disabled the user's focus **location** is still in the area of that button so they can successfully use the keyboard to tab to the next or previous element.

This is also preferred since, as you mentioned, there's nothing to "clear" if the layer is empty.

Reply:
I like the idea of disabling it if we're in the empty state, but I'm not sure that we have a good mechanism for knowing if we're in the empty state. An alternative would be to leave it focused, but just change the focus UI so that it looks better.

Reply:
I would really like to push for disabling. But if that absolutely can't happen, then we should just leave it as is. I'm not sure what you mean by "look better". EuiButtons all have specific and standard focus states that should not be changed.